### PR TITLE
GEDI RH value montly rasterization.

### PR DESCRIPTION
**Pull request by c11**
Simon, this script is used for rasterization of GEDI monthly dataset. It uses UTM tile for the rasterization process. In the current workflow, only a subset of metrics from the L2A HDF5 data files are ingested for rasterization.

### CODE REVIEW

**Comment by c11** [comment]
@googlebot I signed it!

File: dataset/gedi_rasterization.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    shots = []
+    for orbit in orbits:
+        fshots = ee.FeatureCollection(orbit).filterMetadata('quality', 'equals', 1)
+        augmented = fshots.map(updateOrbit)
+        shots.append(augmented)
+
+    shots = ee.FeatureCollection(shots).flatten()
+
+    grid = (ee.FeatureCollection("users/yang/GEETables/GEDI/GEDI_UTM_GRIDS_LandOnly")
```
Comment (simonff) [inline dataset/gedi_rasterization.py:32]: Could you point at the source URL of the UTM grids file?

File: dataset/gedi_rasterization.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        region=box,
+        pyramidingPolicy={".default": 'sample'},
+        scale=25,
+        crs=crs,
+        maxPixels=1e13)
+
+    time.sleep(0.1)
+    task.start()
+
+lookup_file = 'L2A_lookup.txt'
```
Comment (simonff) [inline dataset/gedi_rasterization.py:68]: what is in this file?

File: dataset/gedi_rasterization.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import time
+from ee.batch import Export
+import ast
+
+def make_timestampFromYMD(year, month, day):
+    tdat = datetime.datetime(year, month, day)
+    return time.mktime(tdat.timetuple()) * 1000
+
+def rasterize_GEDI_by_utm_zone(orbits, grid_id, month):
+    props = ["orbit", "track", "beam", "channel", "degrade", "dtime", "quality",
```
Comment (simonff) [inline dataset/gedi_rasterization.py:15]: could you briefly comment why these properties were selected?

File: dataset/gedi_rasterization.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    zone_id = ee.Feature(grid.first()).get('grid_name').getInfo()
+    crs = ee.Feature(grid.first()).get('crs').getInfo()
+
+    iprop = {
+        "month": month,
+        "year": 2019,
+        "grid_id": grid_id,
+        "version": 1,
+        "system:time_start": make_timestampFromYMD(2019, month, 15)
```
Comment (simonff) [inline dataset/gedi_rasterization.py:43]: would system:time_start of the first day of the month and system:time_end of the first day of the next month be more appropriate?

File: dataset/gedi_rasterization.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    count = 389 # total of UTM grids
+    ee.Initialize()
+
+    for grid_id in range(start_id, start_id + count):
+        for k, v in lookup.items():
+            if k != 'May':
+                continue
+
+            print(grid_id, k, '-->', months[k])
+            rasterize_GEDI_by_utm_zone(v, grid_id, months[k])
```
Comment (simonff) [inline dataset/gedi_rasterization.py:100]: could you pass a date object instead? datetime.date(year, month, 1)

File: dataset/gedi_rasterization.py (hunk 1 above)
Comment (c11) [reply dataset/gedi_rasterization.py:32]: https://www.arcgis.com/home/item.html?id=b294795270aa4fb3bd25286bf09edc51

File: dataset/gedi_rasterization.py (hunk 2 above)
Comment (c11) [reply dataset/gedi_rasterization.py:68]: This is a dictionary listing all the point tables for each month. The purpose is to bypass ee.data API call to retrieve data as the tables are static. Please refer to the L2A_lookup.txt.

File: dataset/gedi_rasterization.py (hunk 3 above)
Comment (c11) [reply dataset/gedi_rasterization.py:15]: "orbit", "track", "beam", "channel" are needed to keep the data identity. 

"degrade", "dtime", "quality" are quality flags that were identify as quality flag for L4A and L4B biomass modeling.

File: dataset/gedi_rasterization.py (hunk 4 above)
Comment (c11) [reply dataset/gedi_rasterization.py:43]: Yes.

File: dataset/gedi_rasterization.py (hunk 5 above)
Comment (c11) [reply dataset/gedi_rasterization.py:100]: it is possible. In the current format, the lookup dictionary uses month names as key.

**Review by simonff — APPROVED** [review]

File: dataset/gedi_rasterization.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
+
```
[the remark this answers is not in the document]
Comment (c11) [reply dataset/gedi_rasterization.py:15]: added.

**Review by schwehr — APPROVED** [review]
LGTM

**Pull request merged**